src/Session.hs

```diff
@@ -92,10 +92,10 @@ isSuccess :: Maybe Summary -> Bool
 isSuccess = not . isFailure
 
 parseSummary :: String -> Maybe Summary
-parseSummary r = case reverse $ lines r of
-  x : _ | Just summary <- readMaybe (dropAnsiEscapeSequences x) -> Just summary
-  _ -> Nothing
+parseSummary = findJust . (map $ readMaybe . dropAnsiEscapeSequences) . take 3 . reverse . lines
```

Comment:
My reasoning is that at this point we already know there is a spec, and running a spec will always yield a summary. So I would hope that taking the first thing, starting from the back of the list, that looks like a summary, is fine.